src/ucp/core/ucp_ep.c

```diff
@@ -2437,19 +2494,16 @@ void ucp_ep_do_keepalive(ucp_ep_h ep, ucp_lane_map_t *lane_map)
         ucs_assert(lane < UCP_MAX_LANES);
         /* in case if remote ID is defined, wireup is completed and EP is
          * in connect-to-iface mode then use UCP/AM keepalive */
-        if (ucp_ep_is_am_keepalive(ep, lane)) {
-            rsc_index = ucp_ep_get_rsc_index(ep, lane);
-            status    = ucp_wireup_msg_send(ep, UCP_WIREUP_MSG_EP_CHECK,
-                                            UCS_BIT(rsc_index), NULL);
-            ucs_assert(status == UCS_OK);
+        rsc_index = ucp_ep_get_rsc_index(ep, lane);
+        ucs_assert((rsc_index != UCP_NULL_RESOURCE) ||
+                   (lane == ucp_ep_get_cm_lane(ep)));
+
+        status = ucp_ep_do_uct_ep_keepalive(ep, ep->uct_eps[lane], rsc_index, 0,
+                                            NULL);
+        if (status != UCS_ERR_NO_RESOURCE) {
             *lane_map &= ~UCS_BIT(lane);
-        } else {
-            /* coverity[overrun-local] */
-            status = uct_ep_check(ep->uct_eps[lane], 0, NULL);
-            if (status == UCS_OK) {
-                *lane_map &= ~UCS_BIT(lane);
-            } else if (status != UCS_ERR_NO_RESOURCE) {
-                *lane_map &= ~UCS_BIT(lane);
+
+            if (ucs_unlikely(status != UCS_OK)) {
```

Comment:
this is slowpath, no need for unlikely
also maybe like this
if (status == NO_RESOURCE)
 continue
else if (status != OK)
 ucs_warn(..)